torch/_inductor/codegen/wrapper.py

```diff
@@ -1897,6 +1915,15 @@ def use_thread_local_cached_output_tensor(idx, output):
                 )
             for idx, output in enumerate(output_refs):
                 if config.aot_inductor.abi_compatible:
+                    output_buffer = V.graph.graph_outputs[idx]
+                    if isinstance(output_buffer, ir.ShapeAsConstantBuffer):
+                        # Need to wrap scalar into tensor as the main function returns a vector of tensors
```

Comment:
HMMMM are you ever considering to change this ABI? I can imagine that returning shapes from AOTInductor models is not that common today and so maybe papering over the ABI problem this way is fine, but when we support dynamic shapes AOTInductor models, I *really* don't want input ints to be passed into the model as tensors, it's soooo unnecessary.

cc @oulgen @bdhirsh

Reply:
cc @chenyang78  who may look into this later.